core/src/main/java/tc/oc/pgm/commands/JoinCommands.java

```diff
@@ -47,7 +47,7 @@ public static void join(
       }
     }
 
-    if (sender.hasPermission(Permissions.JOIN_FORCE) && force) {
+    if (sender.hasPermission(Permissions.JOIN_FORCE) && force && !player.isVanished()) {
```

Comment:
Does this check need to be inside `JoinMatchModule`? I think so, because it should also guard against the picker.